src/test/java/org/jabref/logic/util/io/RegExpBasedFileFinderTests.java

```diff
@@ -80,6 +80,49 @@ void testYearAuthFirstPageFindFiles() throws Exception {
                 result);
     }
 
+    @Test
+    void findAssociatedFilesFindFileContainingBracketsFromBracketedExpression() throws Exception {
+        var bibEntry = new BibEntry().withField(StandardField.TITLE, "Regexp from [A-Z]");
+
+        var extension = Collections.singletonList("pdf");
+        var directory = Collections.singletonList(Path.of(FILES_DIRECTORY));
+        RegExpBasedFileFinder fileFinder = new RegExpBasedFileFinder("[TITLE]\\\\.[extension]", ',');
+
+        List<Path> result = fileFinder.findAssociatedFiles(bibEntry, directory, extension);
+        assertEquals(Collections.singletonList(Path.of("src/test/resources/org/jabref/logic/importer/unlinkedFilesTestFolder/Regexp from [A-Z].pdf")),
+                result);
+    }
+
+    @Test
+    void findAssociatedFilesFindCleamedFileFromBracketedExpression() throws Exception {
+        var bibEntry = new BibEntry().withField(StandardField.JOURNAL, "ACM/IEEE-CS");
+
+        var extension = Collections.singletonList("pdf");
+        var directory = Collections.singletonList(Path.of(FILES_DIRECTORY));
+        RegExpBasedFileFinder fileFinder = new RegExpBasedFileFinder("[JOURNAL]\\\\.[extension]", ',');
+
+        List<Path> result = fileFinder.findAssociatedFiles(bibEntry, directory, extension);
+        assertEquals(Collections.singletonList(Path.of("src/test/resources/org/jabref/logic/importer/unlinkedFilesTestFolder/ACM_IEEE-CS.pdf")),
```

Comment:
I know you didn't started this, but it's bad practice to create (empty) files just for the tests. Its better and easier to use a temporary folder and create the files just for the test: https://junit.org/junit5/docs/current/user-guide/#writing-tests-built-in-extensions-TempDirectory

Reply:
I guess I should have known that existed 😬 
I'll look into it tomorrow, I doubt I am the only one that "learns" from the test-files.

Reply:
The TempDir extension is new in Junit5 if I'm not mistaken so it's not surprising that the code is not updated yet. Thanks for looking into it.

Reply:
You can take a look at the RenamePdfCleanupTest for using this

Reply:
Thank you both for the links! This should be addressed now.